src/sdk/object_sdk.js

```diff
@@ -245,18 +271,11 @@ class ObjectSDK {
         }
         // check for a specific bucket
         if (bucket && req.op_name !== 'put_bucket') {
-            // ANONYMOUS: cannot work without bucket, cannot work on namespace bucket (?)
+            // ANONYMOUS: cannot work without bucket.
+            // Return if the acount is anonymous
+            const is_anon = !(token && token.access_key);
+            if (is_anon) return;
             const ns = await this.read_bucket_sdk_namespace_info(bucket);
-            if (!token) {
-                // TODO: Anonymous access to namespace buckets not supported
-                if (ns) {
```

Comment:
anonymous access is still not supported on containerized environments, no?
IIRC this should not be removed, but adding the NC env check to the condition